### PR TITLE
#35 fill pending pool

### PR DESCRIPTION
**Pull request by drehtisch**

### CODE REVIEW

File: NationStatesAPIBot/NationStatesAPIBot/Manager/NationManager.cs [hunk 1]
```diff
@@ -30,6 +40,22 @@ public static int GetNationCountByStatusName(string name)
...
+                    n => n.Name == name 
+                    && n.Status.Name == "pending" 
+                    && n.Status.Name == "skipped" 
+                    && n.Status.Name == "send" 
+                    && n.Status.Name == "failed"
+                    );
+                return nation != null;
+            }
+        }
+
```
Comment (bion33) [inline NationStatesAPIBot/NationStatesAPIBot/Manager/NationManager.cs:59]: Method name says "Or", but you're using &&?

Comment (drehtisch) [reply NationStatesAPIBot/NationStatesAPIBot/Manager/NationManager.cs:59]: Just found and fixed by myself :)

File: NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs [hunk 2]
```diff
@@ -53,6 +53,11 @@ public RecruitmentService(ILogger<RecruitmentService> logger, IOptions<AppSettin
             _apiService = apiService;
             _dumpDataService = dumpDataService;
             defaulEventId = LogEventIdProvider.GetEventIdByType(LoggingEvent.APIRecruitment);
+            _rnd = new Random();
+            if (_config.EnableRecruitment)
+            {
+                RecruitmentStatus = "Disabled";
+            }
```
Comment (bion33) [inline NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs:60]: If EnableRecruitment status is Disabled. That doesn't seem right when just reading it... if it's not a mistake and there's a particular reason for it, could you explain it perhaps in a comment?

Comment (drehtisch) [reply NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs:60]: It is an issue it should be !_config.EnableRecruitment. Thanks for pointing out.

File: NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs [hunk 3]
```diff
@@ -144,23 +150,36 @@ private async Task<bool> IsNationRecruitableAsync(Nation nation, EventId id)
...
             if (nation != null)
             {
-                if (!await DoesNationFitCriteriaAsync(nation) || !await WouldReceiveTelegram(nation.Name))
+                var result = await IsNationRecruitableAsync(nation.Name, id);
+                if (!result)
                 {
-                    _logger.LogDebug(id, LogMessageBuilder.Build(id, $"{nation.Name} does not fit criteria and is therefore skipped"));
                     await NationManager.SetNationStatusToAsync(nation, "skipped");
+                }
+                return true;
```
Comment (bion33) [inline NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs]: This returns true if a nation is skipped, no? Is that intended?

File: NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs [hunk 4]
```diff
@@ -205,16 +223,77 @@ private async Task GetNewNationsAsync()
...
+                    }
+                    var regionId = _rnd.Next(regionsToRecruitFrom.Count);
+                    var region = regionsToRecruitFrom.ElementAt(regionId);
+                    string nationName;
+                    do
+                    {
+                        var nationId = _rnd.Next(region.NATIONNAMES.Count);
+                        nationName = region.NATIONNAMES.ElementAt(nationId);
+                    }
+                    while (await NationManager.IsNationPendingSkippedSendOrFailedAsync(nationName) || !await IsNationRecruitableAsync(nationName, id));
```
Comment (bion33) [inline NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs:254]: Why is this while loop empty? I once used an empty while loop for making my CPU's temperature go as high as possible. Not sure if it does that in c# too though :p

File: NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs (hunk 3 above)
Comment (drehtisch) [reply NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs]: Not it isn't.

File: NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs (hunk 4 above)
Comment (drehtisch) [reply NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs:254]: It should find an nation to be added. So it isn't empty. As IsNationRecruitableAsync will take some time to complete this shouldn't get CPU temperature high.

**Review by bion33 — DISMISSED** [review]

**Review by bion33 — APPROVED** [review]

**Pull request merged**